dipy/reconst/tests/test_fwdti.py

```diff
@@ -0,0 +1,209 @@
+
+    assert_array_almost_equal(Ffwe, GTF, decimal=2)
+    
+
+def test_fwdti_predictions():
+    # single voxel case
+    # test funtion
+    gtf = 0.50  #ground truth volume fraction
+    S0=100
+    angles = [(90, 0), (90, 0)]
+    mevals = np.array([[0.0017, 0.0003, 0.0003], [0.003, 0.003, 0.003]])
+    S_conta, peaks = multi_tensor(gtab_2s, mevals, S0=S0,
+                                  angles=angles,
+                                  fractions=[(1-gtf) * 100, gtf*100], snr=None)
+    R = all_tensor_evecs(peaks[0])
+    R = R.reshape((9))
+    model_params = np.concatenate(([0.0017, 0.0003, 0.0003], R, [gtf, 100]),
+                                  axis=0)
+    S_pred1 = fwdti_prediction(model_params, gtab_2s)
+    assert_array_almost_equal(S_pred1, S_conta)
+
+    # Testing in model class
+    fwdm = fwdti.FreeWaterTensorModel(gtab_2s, 'WLS')
+    S_pred2 = fwdm.predict(model_params)
+    assert_array_almost_equal(S_pred2, S_conta)
+
+    # Testing in fit class
+    fwefit = fwdm.fit(S_conta)
+    # Adjust simulations according to model parameters (note here testing the
+    # robustness of fit is not the objective)
+    mevals_ad = np.array([fwefit.model_params[0:3], [0.003, 0.003, 0.003]])
+    angles_ad = fwefit.model_params[3:-2].reshape(3, 3)
+    gtf_ad = fwefit.model_params[-2]
+    S0 = fwefit.model_params[-1]
+    S_conta_ad, peaks = multi_tensor(gtab_2s, mevals_ad, S0=S0,
+                                     angles=angles_ad,
+                                     fractions=[(1-gtf_ad) * 100, gtf_ad*100],
+                                     snr=None)
+    S_pred3 = fwefit.predict(gtab_2s)
+    assert_array_almost_equal(S_pred3, S_conta_ad, decimal=5)
+
+    # Multi voxel simulation
+    S_pred1 = fwdti_prediction(model_params_mv, gtab_2s)
+    assert_array_almost_equal(S_pred1, DWI)
+    S_pred2 = fwdm.predict(model_params_mv)
+    assert_array_almost_equal(S_pred2, DWI)
+
+
+def test_fwdti_errors():
+    # 1st error - if a unknown fit method is given to the FWTM
+    assert_raises(ValueError, fwdti.FreeWaterTensorModel, gtab_2s,
+                  fit_method="pKT")
+    # 2nd error - if min_signal is negative
+    assert_raises(ValueError, fwdti.FreeWaterTensorModel, gtab_2s,
+                  min_signal=-1)
+    # 3rd error - if incorrect mask is given
+    fwdtiM = fwdti.FreeWaterTensorModel(gtab_2s)
+    incorrect_mask = np.array([[True, True, False], [True, False, False]])
+    assert_raises(ValueError, fwdtiM.fit, DWI, mask=incorrect_mask)
+
+    # Testing the correct usage
+    fwdtiM = fwdti.FreeWaterTensorModel(gtab_2s, min_signal=1)
+    correct_mask = np.zeros((2, 2, 2))
+    correct_mask[0, :, :] = 1;
+    correct_mask = correct_mask > 0
+    fwdtiF = fwdtiM.fit(DWI, mask=correct_mask)
+    assert_array_almost_equal(fwdtiF.fa, FAref)
+    assert_array_almost_equal(fwdtiF.f, GTF)
+
+    # 4th error - if a sigma is selected by no value of sigma is given for
+    # in the non-linear approach to performe restore
```

Comment:
Typo: "performe" => "perform"

In general, maybe: "... if weighting is set to 'sigma', but no value of sigma is given (in case RESTORE is used)."